whois-api/src/main/java/net/ripe/db/whois/api/mail/dequeue/MessageService.java

```diff
@@ -58,21 +59,24 @@ public EmailMessageInfo getAutomatedFailureMessageInfo(final MimeMessage message
         return autoSubmittedMessageParser.parse(message);
     }
 
-    public void verifyAndSetAsUndeliverable(final EmailMessageInfo message) {
-        final List<String> outgoingEmail = outgoingMessageDao.getEmails(message.messageId());
+    public void verifyAndSetAsUndeliverable(final EmailMessageInfo messageInfo) {
+        final List<String> outgoingEmail = outgoingMessageDao.getEmails(messageInfo.messageId());
 
-        if (!isValidMessage(message, outgoingEmail)) {
+        if (!isValidMessage(messageInfo, outgoingEmail)) {
             return;
         }
 
-        LOGGER.debug("Undeliverable message-id {} email {}", message.messageId(), StringUtils.join(message.emailAddresses(), ", "));
-        message.emailAddresses().forEach(email -> {
+        LOGGER.debug("Undeliverable message-id {} email {}", messageInfo.messageId(), StringUtils.join(messageInfo.emailAddresses(), ", "));
+        for (String email : messageInfo.emailAddresses()) {
             try {
-                emailStatusDao.createEmailStatus(email, EmailStatusType.UNDELIVERABLE);
+                emailStatusDao.createEmailStatus(email, EmailStatusType.UNDELIVERABLE, messageInfo.message());
```

Comment:
Shoudl the existing entry needs to be updated with latest messageInfo ? Why it is undeliverable reason can be different?

Reply:
We can not have two same undeliverable emails
If an email is already undeliverable we don't seen another email to it?